src/chapters/shell-intro/intro.rst

```diff
@@ -0,0 +1,132 @@
+
+Throughout this course we will explore both Linux with BASH and Windows with PowerShell. Although PowerShell *can be used* on Linux machines and BASH *can be used* on Windows machines (through the Linux subsystem) we will defer to the default Shells for each of them depending on which machine we are working with.
+
+Terminal Emulators
+===================
+
+Years ago humans interacted with computers using Terminal screens which translated keyboard input signals into commands and displayed output as plain text. Today the CLI uses programs called **Terminal Emulators** which *emulate* the behavior of the original Terminals. 
+
+Terminal Emulators are GUI applications that are used to interact with the CLI Shell of the machine. Because they are emulated in a GUI window they allow some limited use of the mouse for things like copying, pasting and other conveniences.
+
+.. admonition:: note
+
+   We will refer to the Terminal Emulator programs as Terminals going forward for brevity.
+
+Shell REPL
+----------
+
+A REPL is a **R**\ead **E**\valuate **P**\rint and **L**\oop environment for interacting with a Shell.
+
+A REPL environment first presents a **prompt** for the user to input a command. It then **R**\eads the command that is entered. Once it has parsed the input it **E**\valuates the command to compute or perform the requested task. Any output from the command (a response from the OS or another program) is then **P**\rinted out in the Terminal. Finally, the process repeats itself by **L**\ooping back to the prompt for the next command to be entered.
+
+When you open a Terminal application a REPL of the Login Shell will begin automatically. While we will primarily be working with the BASH and PowerShell REPLs they also exist for other languages like MySQL, JavaScript (NodeJS) and Python.
+
+.. admonition:: note
+
+   CLI documentation traditionally uses the ``$`` and ``>`` characters to represent the input prompt for BASH and PowerShell respectively. Every line that begins with one of these characters **should be treated as an individual command** to be entered into the Terminal. 
+   
+   Some commands will be the same in both Shells. For others we will make it clear which OS and Shell we are referring to.
+
+   Linux and BASH examples will be labeled as ``Linux/BASH`` and use the ``$`` symbol:
+
+   .. sourcecode:: bash
+      :caption: Linux/BASH
+   
+      $ command
+
+   Windows and PowerShell examples will be labeled as ``Windows/PowerShell`` and use the ``>`` symbol:
+
+   .. sourcecode:: powershell
+      :caption: Windows/PowerShell
+   
+      > command
+
+Gnome Terminal
+--------------
+
+In this class we will be using the `Ubuntu distribution <https://ubuntu.com/>`_ as our Linux OS. We will refer to this as either Ubuntu or Linux interchangeably for simplicity. Ubuntu is one of the more polished distributions and, in the Desktop version, includes a window manager (GUI Shell) called Gnome.
```

Comment:
fact check or remove / replace with WSL Ubuntu Terminal

Reply:
introduce the Windows Terminal?

believe it be integrated with WSL for seamless switching between shells